### PR TITLE
Don't fire sourcedata event with sourceDataType: 'metadata' when a source is removed

### PR DESCRIPTION
**Pull request by jfirebaugh**
Noticed this while working on #5762 -- this was added in #4842, but it seems odd to me to fire this event when the source itself isn't changing. Instead, have the LogoControl listen for the styledata event that's triggered when a source is removed.

### CODE REVIEW

File: src/ui/control/logo_control.js [hunk 1]
```diff
@@ -32,6 +32,7 @@ class LogoControl {
         this._container.appendChild(anchor);
         this._container.style.display = 'none';
 
+        this._map.on('styledata', this._updateLogo);
```
Comment (mollymerp) [inline src/ui/control/logo_control.js]: you'll also need to remove this event listener in the `onRemove` function

File: src/ui/control/logo_control.js [hunk 2]
```diff
@@ -32,6 +32,7 @@ class LogoControl {
...
         this._map.on('sourcedata', this._updateLogo);
```
Comment (mollymerp) [inline src/ui/control/logo_control.js]: I think you are safe to remove this listener all together. Keeping it here and removing the `metadata` condition causes `_updateLogo` to be called on every tile load 🙃

Comment (jfirebaugh) [reply src/ui/control/logo_control.js]: I think this might still be necessary to ensure that the control updates when TileJSON is loaded.

Comment (mollymerp) [reply src/ui/control/logo_control.js]: good catch, thank you! it was working on the debug page because the spritesheet was loading after the tilejson.

**Comment by jfirebaugh** [comment]
Stale; closing for now.

**Pull request closed without merging**